libtenzir/builtins/formats/syslog.cpp

```diff
@@ -419,6 +419,9 @@ struct legacy_message_parser : parser_base<legacy_message_parser> {
         x.tag = std::nullopt;
         x.process_id = std::nullopt;
       }
+      if (x.process_id == "-") {
+        x.process_id = std::nullopt;
+      }
```

Comment:
Should this be hardcoded here? Also, I think this contradicts the example given in the changelog.